server/game/GameActions/ConditionalGameAction.ts

```diff
@@ -0,0 +1,66 @@
+import { GameAction, GameActionProperties } from './GameAction';
+import AbilityContext = require('../AbilityContext');
+
+export interface ConditionalActionProperties extends GameActionProperties {
+    gameAction: GameAction;
+    condition: boolean | ((context: AbilityContext) => boolean);
+    else?: GameAction;
+}
+
+export class ConditionalGameAction extends GameAction {
+    effect = 'do a conditional action';
+
+    constructor(properties: ConditionalActionProperties | ((context: AbilityContext) => ConditionalActionProperties)) {
+        super(properties);
+    }
+
+    getProperties(context: AbilityContext, additionalProperties = {}): ConditionalActionProperties {
+        let properties = super.getProperties(context, additionalProperties) as ConditionalActionProperties;
+        properties.condition = properties.condition || true;
+        return properties;
```

Comment:
This should probably be handled with defaultProperties.  You should put something in here though which sets the two game action default targets to be this actions target